lexer/impl/lexer.cpp

```diff
@@ -85,7 +85,10 @@ Token Lexer::Peek() {
     if (currentToken == nullptr) {
         currentToken = std::make_shared<Token>(scanNext());
     } else if (currentToken->type == TokenType::Eof) {
-        throw "End-of-file already reached";
+        Token t;
+        t.type = TokenType::Eof;
+        return t;
+        // throw "End-of-file already reached";
```

Comment:
1) why not simply return `currentToken`? There is no need to create another object
2) The reason I had this throw is to stop faulty programs from causing our parser to enter an infinite loop. In some places, we keep eating tokens until we find one particular token (especially in cases of errors), and we don't want to always check for EOF due to an error in the source code (unless really necessary). It also makes sense that after we consume the EOF token, there is nothing more to peek